### PR TITLE
drivers: kscan_touch: enable the touch panel for rt1170_evk

### PR DESCRIPTION
**Pull request by CristXu**
Enable the touch panel for the rt1170_evk

Signed-off-by: Crist Xu <crist.xu@nxp.com>

### CODE REVIEW

File: dts/arm/nxp/nxp_rt1170_cm7.dtsi [hunk 1]
```diff
@@ -48,7 +48,8 @@
 		gpio2: gpio@42008000 {
 			compatible = "nxp,imx-gpio";
 			reg = <0x42008000 0x4000>;
-			interrupts = <102 0>, <103 0>;
+			interrupts = <99 0>;
+			label = "GPIO_2";
```
Comment (galak) [inline dts/arm/nxp/nxp_rt1170_cm7.dtsi:52]: remove `label` prop

File: dts/bindings/kscan/goodix,gt911.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+description: GT9xx / GT9xxx capacitive touch panels
+
+compatible: "goodix,gt911"
+
+include: [kscan.yaml, i2c-device.yaml]
+
+properties:
+    int-gpios:
```
Comment (galak) [inline dts/bindings/kscan/goodix,gt911.yaml:11]: rename to `irq-gpios` to match linux binding.

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts [hunk 3]
```diff
@@ -101,8 +102,18 @@
...
-&lpi2c1 {
+&lpi2c5 {
 	status = "okay";
+	pinctrl-0 = <&pinmux_lpi2c5>;
+	pinctrl-names = "default";
+
+	touch_controller: gt911@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		label = "GT911";
```
Comment (galak) [inline boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:113]: remove `label` prop.

File: boards/arm/mimxrt1170_evk/Kconfig.defconfig [hunk 4]
```diff
@@ -33,6 +33,19 @@ endchoice
 
 endif #FLASH
 
+config I2C
```
Comment (galak) [inline boards/arm/mimxrt1170_evk/Kconfig.defconfig:36]: remove `config I2C` bits, those should get dealt with by the Kconfig of the driver itself.

File: boards/arm/mimxrt1170_evk/Kconfig.defconfig [hunk 5]
```diff
@@ -33,6 +33,19 @@ endchoice
...
+	default y if (KSCAN || SENSOR)
+
+if KSCAN
+
+config KSCAN_GT911
```
Comment (galak) [inline boards/arm/mimxrt1170_evk/Kconfig.defconfig:41]: remove `KSCAN_GT911`, will get enabled by Kconfig and devicetree association.

File: drivers/kscan/Kconfig.gt911 [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2020 NXP
+# Copyright (c) 2020 Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig KSCAN_GT911
+	bool "GT9xx / GT9xxx capacitive touch panel driver"
+	depends on I2C
```
Comment (galak) [inline drivers/kscan/Kconfig.gt911:7]: replace `depends on I2C` with:

```
default y
depends on DT_HAS_GOODIX_GT911_ENABLED
select I2C
```

File: dts/arm/nxp/nxp_rt1170_cm7.dtsi (hunk 1 above)
Comment (galak) [reply dts/arm/nxp/nxp_rt1170_cm7.dtsi:52]: This change for IRQs should be in a separate commit.

**Review by galak — CHANGES_REQUESTED** [review]
A few Kconfig and dts related changes.

File: dts/arm/nxp/nxp_rt1170_cm7.dtsi (hunk 1 above)
Comment (dleach02) [reply dts/arm/nxp/nxp_rt1170_cm7.dtsi:52]: @galak, when do you think it is time to add a CI check for the label prop?

Comment (CristXu) [reply dts/arm/nxp/nxp_rt1170_cm7.dtsi:52]: Will change in another commi

File: drivers/kscan/Kconfig.gt911 (hunk 6 above)
Comment (CristXu) [reply drivers/kscan/Kconfig.gt911]: Done

File: boards/arm/mimxrt1170_evk/Kconfig.defconfig (hunk 5 above)
Comment (CristXu) [reply boards/arm/mimxrt1170_evk/Kconfig.defconfig]: Done

File: boards/arm/mimxrt1170_evk/Kconfig.defconfig (hunk 4 above)
Comment (CristXu) [reply boards/arm/mimxrt1170_evk/Kconfig.defconfig]: done

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts (hunk 3 above)
Comment (CristXu) [reply boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts]: done

File: dts/bindings/kscan/goodix,gt911.yaml (hunk 2 above)
Comment (CristXu) [reply dts/bindings/kscan/goodix,gt911.yaml]: done

File: dts/bindings/kscan/goodix,gt911.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+      type: phandle-array
+      required: false
+    rst-gpios:
```
[the remark this answers is not in the document]
Comment (CristXu) [reply dts/bindings/kscan/goodix,gt911.yaml]: done

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts [hunk 8]
```diff
@@ -15,6 +15,7 @@
 
 	aliases {
 		mipi-dsi = &mipi_dsi;
+		kscan0 = &touch_controller;
```
Comment (galak) [inline boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:18]: replace alias `kscan0`, with a chosen node property:

`zephyr,keyboard-scan = &touch_controller;`

**Review by galak — DISMISSED** [review]
minor change to dtsi

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts (hunk 8 above)
Comment (CristXu) [reply boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts]: done

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts [hunk 9]
```diff
@@ -29,6 +29,7 @@
 		zephyr,flash = &is25wp128;
 		zephyr,code-partition = &slot0_partition;
 		zephyr,display = &lcdif;
+		zephyr,keyboard-scan = &touch_controller;
```
Comment (mbolivar-nordic) [inline boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:32]: @CristXu you need to update the table at the bottom of doc/build/dts/api.rst if you're going to add a new chosen node

@galak don't forget to mention this too please

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts (hunk 9 above)
Comment (galak) [reply boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:32]: @mbolivar-nordic its in that table already - https://docs.zephyrproject.org/latest/build/dts/api/api.html#id2

**Review by dleach02 — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
whoops thanks @galak

**Pull request merged**